### PR TITLE
feat: work in progress for community stories and listings

### PR DESCRIPTION
**Pull request by jhung**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Adding community stories section to the theme. Work in progress.

## Notes

* The working URL is localized to my dev environment, so you will have to change it to match you instance.
* Redirections needed to be updated otherwise going to the stories would redirect to the blog.

### CODE REVIEW

File: resources/views/partials/content-page-stories.blade.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      </h2>
+    </header>
+    <div class="pcc-story__meta">
+
+      <div class="card__region">
+        <span class="screen-reader-text">Region: </span>
+        <svg class="icon icon--location" aria-hidden="true" viewBox="0 0 20 20" focusable="false">
+            <use href="{{ App\asset_path('images/location.svg#location') }}" />
+        </svg>
+        @foreach (Page::storyRegions(get_the_ID()) as $region)
```
Comment (greatislander) [inline resources/views/partials/content-page-stories.blade.php:24]: This controller method is fetching data about an individual story, not a page. I'd recommend following the pattern shown here:

1. Create a `trait` for stories: https://github.com/platform-coop-toolkit/coop-library/blob/dev/app/Controllers/Partials/Resource.php
2. In the `Page` and `SinglePccStory` controllers, import the trait with a `use` statement:
    - https://github.com/platform-coop-toolkit/coop-library/blob/dev/app/Controllers/Single.php
3. In the page template, call the methods using `Page::methodName()` and `SinglePccStory::methodName()` respectively:
    - https://github.com/platform-coop-toolkit/coop-library/blob/dev/resources/views/partials/content-lc_resource.blade.php
    - https://github.com/platform-coop-toolkit/coop-library/blob/dev/resources/views/partials/content-single-lc_resource.blade.php

We can go over this if you like. The advantage is you don't need to pass the ID and you can reuse the same method in both contexts.

File: resources/views/partials/content-single-pcc-story.blade.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
+@php $org = get_post_meta (get_the_ID(), 'pcc_story_organization', true) @endphp
```
Comment (greatislander) [inline resources/views/partials/content-single-pcc-story.blade.php:1]: It's best practice to pass data to the template by creating a controller method. In this case, adding a method called `storyOrg` in the SinglePccStory controller would populate a variable called `$story_org`:

```php
public function storyOrg()
{
  return get_post_meta(get_the_ID(), 'pcc_story_organization', true);
}

Alternatively, you could create a static method in the trait (see previous comment) which would handle this for both the Page and Single views but would need to be called (`Page::storyOrg()` etc).

File: resources/views/partials/content-single-pcc-story.blade.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+<article @php post_class('container') @endphp>
+  <header>
+    @include('partials/breadcrumb')
+    <h1 class="entry-title">{{ $org }} - "{!! App::title() !!}"</h1>
+  </header>
+  @include('partials/entry-meta')
+
+  <div class="content" id="content">
+    <div class="pcc-story-video-container">
+    @php echo wp_oembed_get(get_post_meta (get_the_ID(), 'pcc_story_video_link', true)) @endphp
```
Comment (greatislander) [inline resources/views/partials/content-single-pcc-story.blade.php:11]: I'd recommend making this a Controller method.

File: resources/views/partials/content-single-pcc-story.blade.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      @if ($sectorList)
+      <p>{{ __('Sectors', 'pcc') }}</p>
+      {!! $sectorList !!}
+      @endif
+
+      @if ($regionList)
+      <p>{{ __('Regions', 'pcc') }}</p>
+      {!! $regionList !!}
+      @endif
```
Comment (greatislander) [inline resources/views/partials/content-single-pcc-story.blade.php:30]: The assignment isn't needed; the same functionality can be obtained like this:

```php
@if ($sectors)
  // Code goes here.
@endif
```
The `sectors()` and `regions()` methods aren't static so they don't need to be called statically (and in fact shouldn't be); they are automatically assigned to snake-cased variables.

**Comment by jhung** [comment]
Closing this draft PR.

**Pull request closed without merging**